src/AppInstallerCLICore/Commands/ImportCommand.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "ImportCommand.h"
+#include "Workflows/CompletionFlow.h"
+#include "Workflows/ImportExportFlow.h"
+#include "Workflows/InstallFlow.h"
+#include "Workflows/WorkflowBase.h"
+#include "Resources.h"
+
+namespace AppInstaller::CLI
+{
+    using namespace std::string_view_literals;
+
+    std::vector<Argument> ImportCommand::GetArguments() const
+    {
+        return {
+            Argument::ForType(Execution::Args::Type::ImportFile),
+            Argument::ForType(Execution::Args::Type::Exact),
+        };
+    }
+
+    Resource::LocString ImportCommand::ShortDescription() const
+    {
+        return { Resource::String::ImportCommandShortDescription };
+    }
+
+    Resource::LocString ImportCommand::LongDescription() const
+    {
+        return { Resource::String::ImportCommandLongDescription };
+    }
+
+    std::string ImportCommand::HelpLink() const
+    {
+        // TODO: point to correct location
+        return "https://aka.ms/winget-command-import";
+    }
+
+    void ImportCommand::ValidateArgumentsInternal(Execution::Args& execArgs) const
+    {
+        if (!std::filesystem::exists(execArgs.GetArg(Execution::Args::Type::ImportFile)))
+        {
+            // TODO
+            throw CommandException(Resource::String::VerifyFileFailedNotExist, execArgs.GetArg(Execution::Args::Type::ImportFile));
+        }
+    }
+
+    void ImportCommand::ExecuteInternal(Execution::Context& context) const
+    {
+        context <<
+            Workflow::ReadImportFile <<
+            Workflow::SearchPackagesForImport <<
```

Comment:
I would imagine this set of tasks in the flow:
```
Read file
Add source(s) as needed
For each package in import file
    Search for appropriate version to install given inputs
    If viable package version found, proceed to install
    Else, output warning/error
```
Notably, we would *not* attempt to search for everything before installing the first package.

There are certainly complications with adding the sources and the requirement to be admin to do so, but I don't think that will be a blocker.

Reply:
I disagree here. I would expect it to search everything first and tell me if there is anything missing so I can start the import, leave and come back to a full install. If I was installing a bunch of stuff, I would be quite annoyed if I came back to see that it didn't install half of what I wanted because it didn't find it without telling me upfront.

Reply:
Per offline talk, we will not add the sources automatically and we will be validating the file before installing everything.